shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -1209,15 +1211,23 @@ export default {
           // @TODO what if the pool is missing?
           const id = `pool${ ++this.lastIdx }`;
 
-          out.push({
+          const poolData = {
             id,
             remove: false,
             create: false,
             update: true,
             pool:   clone(pool),
             config: config ? await this.$store.dispatch('management/clone', { resource: config }) : null,
             configMissing
-          });
+          };
+
+          // add data to machine pools array
+          out.push(poolData);
+
+          // but we also store the initial data so that we can handle conflicts
+          const poolDataClone = Object.assign({}, poolData);
```

Comment:
`Object.assign` performs a shallow copy, do we intend to do a deep copy? If so, we'll want to consider using `structuredClone` or Lodash's `cloneDeep`.